Feat(MKS): create guide about migration from free to standard (#8379)
ovh/docs: pages/public_cloud/containers_orchestration/managed_kubernetes/migration-from-free-to-standard/guide.en-gb.md

---
title: MKS Free to Standard Migration Guide

> excerpt: Learn how to migrate your OVH MKS cluster from Free to Standard, understand plan differences, and follow best practices for a secure and smooth transition.

Reviewer: ```suggestion
excerpt: Learn how to migrate your OVHcloud MKS cluster from Free to Standard offer, understand plan differences, and follow best practices for a secure and smooth transition.
```

updated: 2025-09-17
---

> ## Objective

Reviewer: ```suggestion
## Objectives
```

> This guide outlines the process of migrating your OVH Managed Kubernetes Service (MKS) cluster from the Free plan to the Standard plan. It provides a step-by-step walkthrough to ensure a secure and efficient transition while minimizing downtime.

Reviewer: ```suggestion
This guide outlines the process of migrating your OVHcloud Managed Kubernetes Service (MKS) cluster from the Free plan to the Standard plan. It provides a step-by-step procedure to ensure a secure and efficient transition while minimizing downtime.
```

We will cover the essential phases of migration, including:

- **Plan Comparison:** Understanding the differences between Free and Standard plans.
- **Backup and Restore:** Using Trilio, CloudCasa, or Velero to safely migrate your data and workloads.
- **Cluster Migration:** Moving your entire cluster or specific namespaces to the Standard plan.
- **Post-Migration Validation:** Ensuring your applications are fully functional and optimized on the Standard plan.

This guide is designed to provide you with the knowledge and best practices needed for a smooth and successful MKS plan migration.

## Requirements

> To successfully migrate your Kubernetes cluster to OVHcloud, ensure you have the following in place:

Reviewer: By reading this sentence, I understand that we will migrate a K8s cluster existing outside OVHcloud into OVHcloud which is not the purpose of this guide, isn't?

- A [Public Cloud project](/pages/public_cloud/public_cloud_cross_functional/create_a_public_cloud_project) in your OVHcloud account.
- **kubectl:** You'll need the kubectl command-line tool installed to interact with your Kubernetes clusters. Refer to the [official Kubernetes documentation](https://kubernetes.io/docs/tasks/tools/) for installation instructions.

## Comparing Free and Standard MKS Plans

> 

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

## Instructions

> 1. **Install, Configure Backup Tool and back up yout cluster**

Reviewer 2: ```suggestion
1. **Install, Configure Backup Tool and back up your cluster**
```

- For Velero, follow our guide [here](/pages/public_cloud/containers_orchestration/managed_kubernetes/backing-up-cluster-with-velero).
- For Trilio, follow our guide [here](/pages/public_cloud/containers_orchestration/managed_kubernetes/backup-and-restore-cluster-namespace-and-applications-with-trilio).
- For CloudCasa, follow our guide [here](/pages/public_cloud/containers_orchestration/managed_kubernetes/backup-and-restore-cluster-using-cloudcasa).

2. **Create your target Kubernetes cluster on OVHcloud**

Log in to the [OVHcloud Control Panel](/links/manager), go to the `Public Cloud`{.action} section and select the Public Cloud project concerned.

Access the administration UI for your OVHcloud Managed Kubernetes clusters by clicking on `Managed Kubernetes Service`{.action} in the left-hand menu and click on `Create a cluster`{.action}.

![Create a cluster](images/creating-a-cluster1.png){.thumbnail}

3. **Pick a flavour and node pool for your new OVHcloud cluster**

    - **Size your worker nodes:** Carefully assess your existing architecture's CPU and RAM requirements and select OVHcloud node flavors that match these specifications.

>     - **Replicate network setup:** Ensure your new cluster's network configuration mirrors your original cluster (e.g., private nodes on a private subnet, dedicated outbound gateway).

Reviewer: ```suggestion
    - **Replicate network setup:** Ensure your new cluster's network configuration mirrors your original cluster (e.g. private nodes on a private subnet, dedicated outbound gateway).
```

>     - **Choose deployment mode:** Select a deployment mode (e.g., 1AZ or 3AZ) based on your fault tolerance needs and high availability requirements.

Reviewer: ```suggestion
    - **Choose deployment mode:** Select a deployment mode (e.g. 1AZ or 3AZ) based on your fault tolerance needs and high availability requirements.
```

> [!primary]
>
> The Standard plan is currently available only in 3AZ regions (Paris). Make sure your migration targets a compatible region to avoid deployment issues.
>

4. **Restore your backups onto the new cluster**

Before migrating your workloads, you need to deploy your chosen backup tool on the new Standard cluster. This ensures that your backups are properly restored and that the migration process can proceed smoothly.

Choose the documentation corresponding to the tool you are using:

> [!tabs]
> **Via Velero**
>> 1. **Deploy Velero on the new cluster**
>>

> >> Install the Velero Helm chart on your newly created OVHcloud Kubernetes cluster by following [this guide](/pages/public_cloud/containers_orchestration/managed_kubernetes/backing-up-cluster-with-velero/#installing velero)

Reviewer: ```suggestion
>> Install the Velero Helm chart on your newly created OVHcloud Kubernetes cluster by following this [guide](/pages/public_cloud/containers_orchestration/managed_kubernetes/backing-up-cluster-with-velero/#installing velero).
```

>> velero restore get
>>
>> # Check overall restore status
>> velero restore describe YOUR_RESTORE_NAME
>>
>> # Check workloads directly in the cluster
>> kubectl get pods --all-namespaces
>> ```
>>
>> Monitor events and logs if some workloads do not start as expected.
>>
> **Via Trilio**
>> follow the step 4 in the **restoring** part of this [guide](/pages/public_cloud/containers_orchestration/managed_kubernetes/backup-and-restore-cluster-namespace-and-applications-with-trilio/#restoring-mysql-qa-helm-release-backup).
>>
> **Via CloudCasa**
>> follow the step 6 of this [guide](/pages/public_cloud/containers_orchestration/managed_kubernetes/backup-and-restore-cluster-using-cloudcasa/#step6-----Run a restore operation).
>>

5. **Validate the Migration**

After restoring your workloads, check that the cluster is fully functional:

- Verify deployments, services, and ingress controllers.
- Test application endpoints to confirm availability.
- Check logs and events for errors or misconfigurations.
- Execute all your application's integration tests on the new target cluster.

6. **Adjust Cluster Resources**

Some workloads may require fine-tuning after the migration:

> - Re-create Load Balancers or reconfigure Ingress if public IPs have changed.

Reviewer: ```suggestion
- Re-create Load Balancers or reconfigure Ingresses if public IPs have changed.
```

- Ensure CRDs, Secrets, and ConfigMaps are properly restored.

7. **Clean Up Old Resources**

Once you’ve validated that everything is running correctly in the Standard cluster:

> - Decommission the source cluster ( Free plan ) to avoid confusion or accidental use.

Reviewer: Instead of avoiding confusion, a Free plan MKS can be used to host testing and/or staging environments. 
Maybe we should rework this sentence to share another message.

> - Remove any unused backup data if no longer needed.

Reviewer: Hum, is it really recommended?

8. **Set up Saving Plans (if needed)**

Explore the option of [OVHcloud Saving Plans](/pages/public_cloud/public_cloud_cross_functional/savings_plans) to optimize your cloud costs. Learn more about the available Saving Plans to determine if they align with your financial strategy.

9. **Post-Migration Best Practices**

- Set up monitoring and alerting to track the health of the new cluster.
- Schedule regular backups with Velero, Trilio, or CloudCasa.
- Document any configuration changes for future reference.

## Tips and best practices

- Operational Best Practices

>   - **Test restores regularly:** Don’t just configure backups—perform periodic test restores to validate recovery procedures.

Reviewer: ```suggestion
  - **Test restores regularly:** Don’t just configure backups and perform periodically restoration test to validate recovery procedures.
```

  - **Use staging environments:** Run your migration first on a non-production cluster to detect potential issues before moving production workloads.
  - **Check quotas and limits:** Make sure your Public Cloud project has sufficient quotas for nodes, storage, and load balancers.
- Security & Compliance
  - **Revalidate RBAC policies:** Ensure that users and service accounts have only the permissions they need in the new cluster.
  - **Rotate secrets and credentials:** After migration, rotate API keys, tokens, and database passwords to reduce risks.

>   - **Audit compliance requirements:** Verify that the new setup still complies with your organization’s security or regulatory requirements (e.g., GDPR, ISO).

Reviewer: ```suggestion
  - **Audit compliance requirements:** Verify that the new setup still complies with your organization’s security or regulatory requirements (e.g. GDPR, ISO).
```

- Performance & Cost
  - **Enable autoscaling:** Configure cluster autoscaler and HPA (Horizontal Pod Autoscaler) to optimize resource usage.
  - **Right-size your nodes:** Review workloads and adjust node pool flavors to balance performance and cost.
  - **Monitor storage usage:** Keep an eye on PVC consumption and adjust storage classes if necessary.
- Reliability
  - **Use 3AZ mode for production:** Ensure high availability by running workloads in a 3AZ cluster.

>   - **Set up alerts:** Integrate with tools like Prometheus, Grafana ...

Reviewer: ```suggestion
  - **Set up alerts:** Integrate with tools like Prometheus, Grafana...
```

We will cover the essential phases of migration, including:

- **Plan Comparison:** Understanding the differences between Free and Standard plans.
- **Backup and Restore:** Using Trilio, CloudCasa, or Velero to safely migrate your data and workloads.
- **Cluster Migration:** Moving your entire cluster or specific namespaces to the Standard plan.
- **Post-Migration Validation:** Ensuring your applications are fully functional and optimized on the Standard plan.

This guide is designed to provide you with the knowledge and best practices needed for a smooth and successful MKS plan migration.

## Requirements

To successfully update your Kubernetes cluster plan within OVHcloud, ensure you have the following prerequisites:

- A [Public Cloud project](/pages/public_cloud/public_cloud_cross_functional/create_a_public_cloud_project) in your OVHcloud account.
- **kubectl:** You'll need the kubectl command-line tool installed to interact with your Kubernetes clusters. Refer to the [official Kubernetes documentation](https://kubernetes.io/docs/tasks/tools/) for installation instructions.

> ## Comparing Free and Standard MKS Plans

Reviewer: Maybe it could be interesting to add a little warning about the fact that at the moment, FloatingIP can't be attached to MKS nodes. This feature will be released in the next few weeks.

While this guide focuses on your current plan, it can be helpful to understand the differences between the Free and Standard plans. The Standard plan offers additional features such as cross-AZ resilience, higher availability SLA, dedicated etcd, and larger maximum cluster size.

> For a complete overview, including a detailed comparison table between Free and Standard, please refer to our [MKS Standard Plan](/pages/public_cloud/containers_orchestration/managed_kubernetes/premium#comparison-plan-array) guide. The table in that guide provides a clear side-by-side comparison of key features.

Reviewer: ```suggestion
For a complete overview, including a detailed comparison table between Free and Standard, please refer to our [MKS Standard Plan](/pages/public_cloud/containers_orchestration/managed_kubernetes/standard#comparison-plan-array) guide. The table in that guide provides a clear side-by-side comparison of key features.
```

>>
>> Example of checking available backups:
>>
>> ```bash
>> velero backup get <backupName>
>> ```
>>
>> Replace <backupName> with the name of your backup.
>>
>> 3. **Restore your backup**
>>
>> Use the Velero CLI to restore resources from a backup:
>>
>> ```bash
>> velero restore create --from-backup <backupName>
>> ```
>>
>> Replace <backupName> with the name of the backup you want to restore.
>>
>> You can add options such as --include-namespaces or --exclude-resources to fine-tune what is restored.
>>
>> 4. **Validate the restore**
>>
>> Once the restore is complete, check that your workloads, services, and configurations are correctly running:
>>
>> ```bash
>> # List the restores
>> velero restore get
>>
>> # Check overall restore status
>> velero restore describe YOUR_RESTORE_NAME
>>
>> # Check workloads directly in the cluster
>> kubectl get pods --all-namespaces
>> ```
>>
>> Monitor events and logs if some workloads do not start as expected.
>>
> **Via Trilio**

> >> follow the step 4 in the **restoring** part of this [guide](/pages/public_cloud/containers_orchestration/managed_kubernetes/backup-and-restore-cluster-namespace-and-applications-with-trilio/#restoring-mysql-qa-helm-release-backup).

Reviewer: ```suggestion
>> Follow the step 4 in the **restoring** part of this [guide](/pages/public_cloud/containers_orchestration/managed_kubernetes/backup-and-restore-cluster-namespace-and-applications-with-trilio/#restoring-mysql-qa-helm-release-backup).
```

>>
> **Via CloudCasa**

> >> follow the step 6 of this [guide](/pages/public_cloud/containers_orchestration/managed_kubernetes/backup-and-restore-cluster-using-cloudcasa/#step6-----Run a restore operation).

Reviewer: ```suggestion
>> Follow the step 6 of this [guide](/pages/public_cloud/containers_orchestration/managed_kubernetes/backup-and-restore-cluster-using-cloudcasa/#step6-----Run a restore operation).
```
